fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/directory/AgilityContextTest.java

```diff
@@ -0,0 +1,217 @@
+                testAgilityContextConcurrentSingleObject(agilityContext, false);
+                context.commit();
+            }
+        }
+        assertLoopThreadsValues();
+    }
+
+    void testAgilityContextOneLongWrite(int loopCount, int sizeLimit, int timeLimit) {
+        final RecordLayerPropertyStorage.Builder insertProps = RecordLayerPropertyStorage.newBuilder()
+                .addProp(LuceneRecordContextProperties.LUCENE_AGILE_COMMIT_SIZE_QUOTA, sizeLimit)
+                .addProp(LuceneRecordContextProperties.LUCENE_AGILE_COMMIT_TIME_QUOTA, timeLimit);
+
+        String ShelSilverstein =
+                "Oh, if you’re a bird, be an early bird\n" +
+                "And catch the worm for your breakfast plate.\n" +
+                "If you’re a bird, be an early early bird--\n" +
+                "But if you’re a worm, sleep late.";
+        try (FDBRecordContext context = openContext(insertProps)) {
+            final AgilityContext agilityContext = AgilityContext.factory(context, true);
+            for (int i = 0; i < loopCount; i++) {
+                byte[] key = Tuple.from(2023, i).pack();
+                byte[] val = Tuple.from(i, ShelSilverstein).pack();
+                agilityContext.set(key, val);
+            }
+            context.commit();
+        }
+        try (FDBRecordContext context = openContext(insertProps)) {
+            final AgilityContext agilityContext = AgilityContext.factory(context, false);
+            for (int i = 0; i < loopCount; i++) {
+                byte[] key = Tuple.from(2023, i).pack();
+                byte[] val = Tuple.from(i, ShelSilverstein).pack();
+                agilityContext.get(key).thenApply(bytes -> {
+                    assertEquals(val, bytes);
+                    return null;
+                });
+            }
+        }
+    }
+
+    @Test
+    void testAgilityContextSizeLimit() {
+        testAgilityContextOneLongWrite(73, 100, 100000);
+    }
+
+    @Test
+    void testAgilityContextTimeLimit() {
+        testAgilityContextOneLongWrite(77, 100000, 10);
+    }
+
+    void napTime(int napTimeMilliseconds) {
+        try {
+            Thread.sleep(napTimeMilliseconds);
+        } catch (InterruptedException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @ParameterizedTest
+    @ValueSource(booleans = {false, true})
+    void testAgilityContextAtomicAttribute(boolean useAgile) {
+        final long prefix = 0x1abc1356; // avoid other tests' data
+        IntStream.rangeClosed(0, threadCount).parallel().forEach(threadNum -> {
+            try (FDBRecordContext context = openContext()) {
+                final AgilityContext agilityContext = AgilityContext.factory(context, useAgile);
+                for (long i = 1700; i < 1900; i += 17) {
+                    final long iFinal = i;
+                    if (threadNum == 0) {
+                        agilityContext.accept(aContext -> {
```

Comment:
Another test that would be good to add, would be to have many parallel writers, here you only have one writer.

Reply:
Good idea! Implemented in both tests (i.e single and multi contexts). 
✅